### PR TITLE
perf_UDT_probes.py:Added test-case for perf UDT instrumentation

### PR DESCRIPTION
**Pull request by naveen538**
By using dtrace and systemtap we are checking perf UDT instrumentation

Signed-off-by: Naveen kumar T <naveet89@in.ibm.com>

### CODE REVIEW

**Comment by naveen538** [comment]
job log:
--------
avocado run perf_UDT_probes.py
JOB ID     : 75a47fdaf9e88035b913efed1b8e3cf1207fad1b
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2019-09-24T15.13-75a47fd/job.log
 (1/1) perf_UDT_probes.py:PerfProbe.test_probe: PASS (0.87 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 1.12 s

File: perf/perf_UDT_probes.py.data/tick-main.c [hunk 1]
```diff
@@ -0,0 +1,20 @@
+#include <stdio.h>
```
Comment (PraveenPenguin) [inline perf/perf_UDT_probes.py.data/tick-main.c]: please have author  signature

File: perf/perf_UDT_probes.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+                        os.path.join(self.workdir, 'tick-main.c'))
+        os.chdir(self.workdir)
+        process.run("dtrace -G -s tick-dtrace.d -o tick-dtrace.o")
+        process.run("dtrace -h -s tick-dtrace.d -o tick-dtrace.h")
+        process.run("gcc -c tick-main.c")
+        process.run("gcc -o tick tick-main.o tick-dtrace.o")
+
+    def test_probe(self):
+        res = process.run("readelf -n tick")
+        if 'NT_STAPSDT' not in res.stdout_text:
```
Comment (PraveenPenguin) [inline perf/perf_UDT_probes.py]: is this is tested on python 3 or avocado upstream

File: perf/perf_UDT_probes.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        '''
+        # Check for basic utilities
+        smm = SoftwareManager()
+        detected_distro = distro.detect()
+        self.distro_name = detected_distro.name
+        deps = ['gcc', 'make']
+        if 'Ubuntu' in self.distro_name:
+            deps.extend(['linux-tools-common', 'linux-tools-%s' %
+                         platform.uname()[2]])
+        elif self.distro_name in ['rhel', 'SuSE', 'fedora', 'centos']:
```
Comment (PraveenPenguin) [inline perf/perf_UDT_probes.py]: have we tested in fedora or centos as well ?

File: perf/perf_UDT_probes.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+from avocado import main
+from avocado.utils import distro, process
+from avocado.utils.software_manager import SoftwareManager
+
+
+class PerfProbe(Test):
+
+    def setUp(self):
+        '''
+        Install the basic packages to support perf and systemtap-sdt-devel.ppc64le
```
Comment (PraveenPenguin) [inline perf/perf_UDT_probes.py]: is this supported on ppc64le system, if yes, please add check for that as well

File: perf/perf_UDT_probes.py (hunk 2 above)
Comment (naveen538) [reply perf/perf_UDT_probes.py]: Yes we tested in python3.

File: perf/perf_UDT_probes.py (hunk 3 above)
Comment (naveen538) [reply perf/perf_UDT_probes.py]: i have tested in RHEL and SUSE. So, removing fedora and centos in code

File: perf/perf_UDT_probes.py (hunk 4 above)
Comment (naveen538) [reply perf/perf_UDT_probes.py]: Its a generic code so not included any specific checks.

File: perf/perf_UDT_probes.py (hunk 3 above)
Comment (PraveenPenguin) [reply perf/perf_UDT_probes.py]: ok

File: perf/perf_UDT_probes.py (hunk 4 above)
Comment (PraveenPenguin) [reply perf/perf_UDT_probes.py]: ah sure maybe than comment need to be rephrased

File: perf/perf_UDT_probes.py (hunk 2 above)
Comment (PraveenPenguin) [reply perf/perf_UDT_probes.py]: ah can you please past the logs

Comment (naveen538) [reply perf/perf_UDT_probes.py]: job log:
--------
avocado run perf_UDT_probes.py
JOB ID     : fc34ae4b74be8a86b693b366b97d8d2f64e52330
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2019-09-25T14.56-fc34ae4/job.log
 (1/1) perf_UDT_probes.py:PerfProbe.test_probe: PASS (0.30 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 0.97 s

File: perf/perf_UDT_probes.py (hunk 4 above)
Comment (naveen538) [reply perf/perf_UDT_probes.py]: yeah done

**Pull request merged**